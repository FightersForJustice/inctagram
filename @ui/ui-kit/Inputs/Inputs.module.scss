@use '../../../@ui/design/settings/colors.scss' as colors;
@import url('https://fonts.googleapis.com/css2?family=Inter:wght@100;200;300;400;500;600;700;800;900&display=swap');

.formField {
  display: flex;
  align-items: baseline;
  justify-content: space-between;
  padding-bottom: 25px;
}

.input {
  font-family: 'Inter';
  font-size: 16px;
  font-style: normal;
  font-weight: 400;
  line-height: 24px;
  width: 100%;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  border-radius: 2px;
  border: 1px solid colors.$dark-100;
  color: colors.$light-900;
  background-color: colors.$dark-500;
  padding: 5px 11px;
  margin-top: 4px;

  &:hover {
    border-color: colors.$light-900;
  }

  &:focus {
    border-color: colors.$light-100;
    color: colors.$light-100;
    background-color: colors.$dark-300;
  }

  &:disabled {
    border-color: colors.$dark-100;
    color: colors.$dark-100;

    &::placeholder {
      color: colors.$dark-100;
    }
  }

  &:active {
    border-color: colors.$light-500;
  }

  &:-webkit-autofill,
  &:-webkit-autofill:hover, 
  &:-webkit-autofill:focus {
    -webkit-text-fill-color: colors.$light-900;
    -webkit-box-shadow: 0 0 0px 1000px colors.$dark-500 inset;
  }
  &:-webkit-autofill:focus {
    -webkit-text-fill-color: colors.$light-100;
    -webkit-box-shadow: 0 0 0px 1000px colors.$dark-300 inset;
  } 
  

  &::placeholder {
    color: colors.$light-900;
  }

  &Error {
    border: 1px solid colors.$danger-500;
  }
}

.errorMessage {
  color: colors.$danger-500;
  margin: 4px 0px 0px 5px;
  float: left;
  text-align: left;
  font-size: 13px;
}

.label {
  float: left;
  color: colors.$light-900;
  font-size: 16px;
  font-weight: 400;

  &Disabled {
    color: colors.$dark-100
  }
}

.eyeButton {
  position: absolute;
  right: 10px;
  top: 31px;
  color: white;
  cursor: pointer;
  filter: invert(0.9);
  transform: scale(1.2);
}