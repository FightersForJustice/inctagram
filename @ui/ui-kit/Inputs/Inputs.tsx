import style from './Inputs.module.scss'
import React, { useState } from 'react'
import LightEyeOpen from '../../../public/icons/lightEyeOpen.svg'
import LightEyeClosed from '../../../public/icons/lightEyeClosed.svg'
import classNames from 'classnames'

interface IMainInputProps extends React.HTMLAttributes<HTMLInputElement> {
  validation?: object
  type?: string
  label?: string
  disabled?: boolean
  errormessages?: Array<string | undefined>
}

type FormInputType = {
  validation?: object
  label: string
  id: string
  name: string
  value: string | number | readonly string[] | undefined //fix
  onChange: (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => void
}

type FormTextareaType = FormInputType & {}

export const MainInput: React.FC<IMainInputProps> = ({ validation, ...props }) => {
  const { label, id, disabled } = props
  const errorMessage = props?.errormessages?.find((e) => e !== undefined)
  return (
    <>
      {label && (
        <label htmlFor={id} className={classNames(style.label, { [style.labelDisabled]: disabled })}>
          {label}
        </label>
      )}
      <input className={classNames(style.input, { [style.inputError]: errorMessage })} {...props} {...validation} />
      {errorMessage && <span className={style.errorMessage}>{errorMessage}</span>}
    </>
  )
}

export const PasswordInput: React.FC<IMainInputProps> = ({ ...props }) => {
  const [isPasswordShown, setIsPasswordShown] = useState(false)
  const labelMargin = props.label ? '0px' : '-20px'
  const {disabled} = props
  return (
    <>
      <div style={{ position: 'relative', width: '100%' }}>
        <span className={classNames(style.eyeButton, {[style.eyeButton_disabled]: disabled})} style={{marginTop: labelMargin}} onClick={() => !disabled && setIsPasswordShown((prev) => !prev)} role='button'>
          {isPasswordShown ? <LightEyeOpen /> : <LightEyeClosed />}
        </span>
      </div>
      <MainInput {...props} type={isPasswordShown ? 'text' : 'password'} />
    </>
  )
}

export const FormInput: React.FC<FormInputType> = ({ onChange, label, id, name, value }) => {
  return (
    <fieldset>
      <label className={style.label} htmlFor={id}>
        {label}
      </label>
      <input className={classNames(style.input, style.input_dark)} id={id} name={name} value={value} onChange={onChange} />
    </fieldset>
  )
}
export const FormTextarea: React.FC<FormTextareaType> = ({ onChange, label, id, name, value }) => {
  return (
    <fieldset>
      <label className={style.label} htmlFor={id}>
        {label}
      </label>
      <textarea className={classNames(style.input, style.input_dark)} id={id} name={name} value={value} onChange={onChange} />
    </fieldset>
  )
}
