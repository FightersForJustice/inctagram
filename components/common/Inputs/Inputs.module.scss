@import url('https://fonts.googleapis.com/css2?family=Inter:wght@100;200;300;400;500;600;700;800;900&display=swap');
/*  ?  */


.FormField {
  display: flex;
  align-items: baseline;
  justify-content: space-between;
  padding-bottom: 25px;
}

.Input {
  width: 100%;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  border-radius: 4px;
  border: 1px solid var(--dark-100, #4c4c4c);
  font-size: 15px;
  color: white;
  background-color: var(--black-a5);
  box-shadow: 0 0 0 1px var(--black-a9);
}

.Input:hover {
  box-shadow: 0 0 0 1px black;
}

.Input:focus {
  box-shadow: 0 0 0 2px black;
}

.Input::selection {
  background-color: var(--black-a9);
  color: white;
}

.Input {
  padding: 0 10px;
  height: 35px;
  line-height: 1;
}

.Label {
  float: left;
  color: #8d9094;
  font-size: 16px;
  font-weight: 400;
}

.EyeButton {
  position: absolute;
  right: 10px;
  font-size: 22px;
  color: white;
  cursor: pointer;
}
