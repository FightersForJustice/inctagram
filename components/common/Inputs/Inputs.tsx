import style from './Inputs.module.scss'
import { useState } from 'react'

interface IMainInputProps extends React.HTMLAttributes<HTMLInputElement> {
  validation?: object
  type?: string
  label?: string
}

export const MainInput: React.FC<IMainInputProps> = ({ validation, ...props }) => {
  return (
    <>
      {props.label && (
        <label htmlFor={props.id} className={style.Label}>
          {props.label}
        </label>
      )}
      <input className={style.Input} {...props} {...validation} />
    </>
  )
}

export const PasswordInput: React.FC<IMainInputProps> = ({ ...props }) => {
  const [isPasswordShown, setIsPasswordShown] = useState(false)
  return (
    <div style={{ position: 'relative' }}>
      <MainInput {...props} type={isPasswordShown ? 'text' : 'password'} />
      <span className={style.EyeButton} onClick={() => setIsPasswordShown((prev) => !prev)}>
        {isPasswordShown ? <span>&#8413;</span> : <span>&#8416;</span>}
      </span>
    </div>
  )
}
