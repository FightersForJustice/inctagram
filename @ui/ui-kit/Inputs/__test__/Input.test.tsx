import { render, cleanup, screen, fireEvent } from '@testing-library/react'
import { MainInput } from '../Inputs'
describe('Input renders correctly', () => {
  afterEach(cleanup)

  const testError = 'test error message'
  const testLabel = 'test label'

  it('should render default correctly', () => {
    const { asFragment } = render(<MainInput />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render error correctly', () => {
    const { asFragment } = render(<MainInput errormessages={[testError]} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render disabled correctly', () => {
    const { asFragment } = render(<MainInput disabled />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render with label correctly', () => {
    const { asFragment } = render(<MainInput label={testLabel} />)
    expect(asFragment()).toMatchSnapshot()
  })
})

describe('Input', () => {
  afterEach(cleanup)

  const testError = 'test error message'
  const testLabel = 'test label'
  const testId = 'test id'
  const testPlaceholder = 'test placeholder'

  it('renders correctly', () => {
    const { getByPlaceholderText } = render(<MainInput placeholder={testPlaceholder} />)

    const inputElement = getByPlaceholderText(testPlaceholder)

    expect(inputElement).toBeInTheDocument()
  })

  it('should be disabled when disabled prop is true', () => {
    const { getByPlaceholderText } = render(<MainInput placeholder={testPlaceholder} disabled />)

    const inputElement = getByPlaceholderText(testPlaceholder)

    expect(inputElement).toBeDisabled()
  })

  it('should display error', () => {
    const { getByPlaceholderText } = render(<MainInput placeholder={testPlaceholder} errormessages={[testError]} />)

    const inputElement = getByPlaceholderText(testPlaceholder)

    expect(inputElement).toHaveClass('inputError')
  })

  it('should focus input when it`s clicked', () => {
    const { getByPlaceholderText } = render(<MainInput placeholder={testPlaceholder} errormessages={[testError]} />)

    const inputElement = getByPlaceholderText(testPlaceholder)
    fireEvent.click(inputElement)

    expect(inputElement).toBeEnabled()
  })

  it('should focus input when label clicked', () => {
    const { getByPlaceholderText } = render(<MainInput id={testId} label={testLabel} placeholder={testPlaceholder} />)

    const labelElement = screen.getByText(testLabel)
    const inputElement = getByPlaceholderText(testPlaceholder)
    fireEvent.click(labelElement)

    expect(inputElement).toBeEnabled()
  })

  it('should display error message on screen', () => {
    const { getByText } = render(<MainInput placeholder={testPlaceholder} errormessages={[testError]} />)
    const errorElement = getByText(testError)

    expect(errorElement).toBeInTheDocument()
  })
})
