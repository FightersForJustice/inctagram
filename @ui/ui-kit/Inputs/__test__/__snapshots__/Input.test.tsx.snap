// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly should render default correctly 1`] = `
<DocumentFragment>
  <input
    class="input"
  />
</DocumentFragment>
`;

exports[`Input renders correctly should render disabled correctly 1`] = `
<DocumentFragment>
  <input
    class="input"
    disabled=""
  />
</DocumentFragment>
`;

exports[`Input renders correctly should render error correctly 1`] = `
<DocumentFragment>
  <input
    class="input inputError"
    errormessages="test error message"
  />
  <span
    class="errorMessage"
  >
    test error message
  </span>
</DocumentFragment>
`;

exports[`Input renders correctly should render with label correctly 1`] = `
<DocumentFragment>
  <label
    class="label"
  >
    test label
  </label>
  <input
    class="input"
    label="test label"
  />
</DocumentFragment>
`;
